magento/pwa-studio
# [bug] Use gql persistence for checkout step between refreshes.

## Description

The checkout step is used to inform guest checkout flow which step we're in and to prevent early submission before all the data necessary is present. We use gql to store and persist the `checkoutStep` across refreshes so that a user returning to checkout is presented at the point in time they left off before.

## Related Issue





Closes [PWA-523](https://jira.corp.magento.com/browse/PWA-523).

## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps


1. Go to the cart page.
2. Check boxes in gift options and add a message. 
3. Refresh the page and check that section. Your selections/message should persist.
4. Go to checkout and refresh the page after you submit each step. The page should render the step you were just on instead of cascading from each previous step.
5. Delete `apollo-cache-persist` entry ONLY from local storage and refresh. The page should render what looks like a cascade while the effects trigger each `onSave` after fetching data. As there is no way to store nonce info for the payment info summary view, this step will appear as if you never submitted it now.
6. Log in/out. All `$Cart` should be deleted and the `Cart` entry should be recreated with a new id.

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.
